test/unit/org/apache/cassandra/io/filesystem/ListenableFileSystem.java

```diff
@@ -796,18 +832,30 @@ else if (mode == MapMode.READ_WRITE)
                     long pos = position;
                     try
                     {
-                        while (local.hasRemaining())
+                        // the channel could be closed... so always create a new channel to avoid this problem
+                        try (FileChannel channel = provider().newFileChannel(path, Set.of(StandardOpenOption.WRITE)))
```

Comment:
I am finding cases where the journal file is deleted *before* we finish writing to it... which is fine from a FS point of view... so to better handle that without failing due to the channel being closed, I moved to always opening a new channel for the write... if the file doesn't exist we just no-op (which is the behavior on real FS anyways